### PR TITLE
Modifier key v-on modifier for keyboard event && Spelling correct &&…

### PR DESCRIPTION
**Pull request by KingMario**
Modifier keys v-on modifier for keyboard event.

Spelling correct.

Invalid v-on modifier key alias warning.

### CODE REVIEW

**Comment by KingMario** [comment]
https://github.com/vuejs/vue/issues/4347 addressed @defcc

**Comment by defcc** [comment]
@KingMario I think you could check the special keycode alias in the `_k` function and introduce the warning once.

**Comment by KingMario** [comment]
@defcc What if `_k` is never called? I guess a global triggered warning can educate the user whenever a built-in alias is defined.

Though the watcher in a dedicated Vue instance solution is no so neat.

File: src/core/index.js [hunk 1]
```diff
@@ -1,9 +1,38 @@
 import Vue from './instance/index'
 import { initGlobalAPI } from './global-api/index'
 import { isServerRendering } from 'core/util/env'
+import { warn } from 'core/util/index'
+import { modifierCode } from '../compiler/codegen/events'
```
Comment (yyx990803) [inline src/core/index.js]: Modules under `/core` should not import from modules under `/compiler`. The only coupling between core and compiler happens in entry files.

File: src/core/index.js [hunk 2]
```diff
@@ -1,9 +1,38 @@
...
 
+for (const key in modifierCode) {
+  Vue.config.keyCodes[key] = undefined
+}
+
+new Vue({
+  data: {
+    config: Vue.config
+  },
+  watch: {
```
Comment (yyx990803) [inline src/core/index.js]: this is definitely less than ideal...

**Comment by yyx990803** [comment]
Implemented the warning with Proxy: e8d6bd9

**Comment by KingMario** [comment]
受教了，谢谢大神！

**Pull request closed without merging**